src/Contexts/PrintLoadingContext.tsx

```diff
@@ -33,50 +33,64 @@ export enum PrintLoadingActions {
-        logger.debug('Promise received');
-        // You must use the other action to increase expected counts.
-        return {expected: state.expected+1, arr: [...state.arr, action.payload]};
-    case PrintLoadingActions.ADD_PROMISE:
-        logger.debug('Promise received');
-        // You must use the other action to increase expected counts.
-        return {expected: state.expected, arr: [...state.arr, action.payload]};
-    case PrintLoadingActions.PRINT:
-        if (state.expected !== action.expected) {
-            logger.debug(`Skipping early print call from ${action.expected} -> ${state.expected}`);
-            return state;
-        }
-        logger.debug(`Printing because ${state.expected} == ${action.expected}`);
-        print();
-        return state;
-    }
-    return state;
-};
-
 export const usePrintLoadingContext = () => React.useContext(PrintLoadingContext);
 
 export const PrintLoadingProvider: React.FC<Props>  = ({ children }) => {
-    const [state, dispatch] = useReducer(reducer, {expected: 1, arr: []});
+    const timeout = React.useRef<NodeJS.Timeout | null>(null);
+    const state: PrintLoadingState = {expected: 0, arr: []};
 
-    useEffect(()=>{
-        if (state.expected > 1 && state.arr.length >= state.expected && Promise.allSettled(state.arr)) {
-            logger.debug(`${state.expected} components have loaded, printing in two seconds.`);
+    const reducer = (state: PrintLoadingState, action: PrintLoadingDispatch): PrintLoadingState => {
+        switch(action.type) {
+        case PrintLoadingActions.RESET_EXPECTED_COUNT:
+            logger.debug('PrintLoadingContext: Printing state reset.', state.expected, state.arr.length);
+            state.expected = 1;
+            state.arr = [];
+            break;
+        case PrintLoadingActions.ADD_EXPECTED_PROMISE_COUNT:
+            state.expected = state.expected + action.expected;
+            break;
+        case PrintLoadingActions.ADD_PROMISE_INCR_EXPECTED:
+            logger.debug('PrintLoadingContext: Promise received', state.expected, state.arr.length);
+            // You must use the other action to increase expected counts.
+            state.expected = state.expected+1;
+            state.arr.push(action.payload);
+            break;
+        case PrintLoadingActions.ADD_PROMISE:
+            logger.debug('PrintLoadingContext: Promise received', state.expected, state.arr.length);
+            // You must use the other action to increase expected counts.
+            state.arr.push(action.payload);
 
-            setTimeout(() => {
-                dispatch({
-                    type: PrintLoadingActions.PRINT,
-                    expected: state.expected,
-                });
-            }, 10000);
+            // If a promise was added and there's already a timeout waiting to be evaluated, remove it.
+            if (!_.isNull(timeout.current)) {
+                clearTimeout(timeout.current);
```

Comment:
After clearing the timeout shouldn't you set it to null?
Clearing a timeout multiple times doesn't have any adverse effects I don't think but just to have a source of truth